README.md

```diff
@@ -2,3 +2,209 @@
+
+- sample test scripts to easily run experiments and test both the accountability algorithm and the interaction between monitor and validators.
+
+## Context and overview
+Documentation describing the context and the theoretical background can be found in the [docs](docs) folder.   
+
+## Architecture
+
+The project is organized in several files and packages in order to guarantee modularity and extendability and maintain at the same time a clear and simple structure.
+
+The monitor (monitor package) is the entity responsible to run the accountability algorithm. It takes as input parameters:
+
+- a path of a yaml configuration file which are necessary to initialize and configure the algorithm
+
+- an (optional) path of a file that can be generated to provide detailed information about the whole execution and, especially, of the accountability algorithm
+
+The monitor is responsible to open the connection with all the validators and initialize the request of the message logs.
+
+Validators are simple processes that listen to a given port and each one has its own messages logs that are the result of the execution of the Tendermint consensus protocol. Message logs and listening port are initialized through a configuration file (different for every validator) along with a unique identifier.
+Every message in the configuration file must be specified with all the corresponding information associated with it (type, round, height, senderId, possible justifications).
+
+The monitor will use the connection library to request message logs from all the addresses given in the config file. It will wait for responses from each validator and, as soon as a packet arrives, it will store it and send it to the main thread for running the algorithm if enough messages have been received until that time.
+The request will be repeated after a timeout is expired and if the message received is not valid. If the validator closes a connection or crashes, the monitor will stop waiting for packets from it and will notify the main thread about the failure in the reception.
+
+The validator, after receiving a valid request packet, will response back if it will have the message logs requested. Otherwise, it will just ignore the request and will not answer the monitor. Optionally, it's possible to configure a response to immediately inform the monitor about the missing log in order to save resources.
+
+The main accountability algorithm is implemented in the accountability package and is described in details in documentation files of the docs folders. Please refer to for a theoretical background or for implementation-specific details.
+
+The connection library implemented in this project wraps the well-known golang/conn library and provides some abstractions to establish a TCP connection, send and receive TCP packets, serialize and de-serialize messages and listen to a specific port.
```

Comment:
For completeness, might be a good idea to add a URL reference to the go pkg here.

Reply:
Fixed